src/panel/direction/MobileRoadMapPreview.jsx

```diff
@@ -1,63 +1,98 @@
-  static propTypes = {
-    steps: PropTypes.array.isRequired,
-    onClose: PropTypes.func.isRequired,
-  }
-
-  state = {
-    currentStep: 0,
-  }
-
-  componentDidMount() {
-    this.zoomToCurrentStep();
-    fire('move_mobile_geolocation_button', 65);
-  }
-
-  zoomToCurrentStep = () => {
-    fire('zoom_step', this.props.steps[this.state.currentStep]);
-  }
-
-  previous = () => {
-    this.setState(
-      prevState => ({ currentStep: Math.max(prevState.currentStep - 1, 0) }),
-      this.zoomToCurrentStep
-    );
-  }
+import classnames from 'classnames';
+import { Button } from '../../components/ui';
+
+const MobileRoadMapPreview = ({
+  steps,
+  onClose,
+}) => {
+  const [currentStep, setCurrentStep] = useState(0);
+  const [expanded, setExpanded] = useState(false);
+  const stepsRef = React.createRef();
+
+  const scroll = () => {
 
-  next = () => {
-    this.setState(
-      prevState => ({ currentStep: Math.min(prevState.currentStep + 1, this.props.steps.length) }),
-      this.zoomToCurrentStep
+    const newStep = Math.floor(
+      // Divide the step container's scrollLeft up to the middle of the screen with the size of a step
+      // to determine which step is present at the middle of the screen
+      (stepsRef.current.scrollLeft + window.innerWidth / 2) / (window.innerWidth - 70 + 12)
     );
-  }
-
-  render() {
-    const currentStep = this.props.steps[this.state.currentStep];
-
-    return <React.Fragment>
-      <div className="itinerary_mobile_step">
-        <CloseButton onClick={this.props.onClose} />
-        <RoadMapStep step={currentStep} />
-      </div>
-
-      <div className="itinerary_mobile_step_buttons">
-        <button
-          className="itinerary_mobile_step_button icon-chevron-left"
-          disabled={this.state.currentStep === 0}
-          onClick={this.previous}
-        />
-        <button
-          className="itinerary_mobile_step_button icon-chevron-right"
-          disabled={this.state.currentStep === this.props.steps.length - 1}
-          onClick={this.next}
-        />
-      </div>
-    </React.Fragment>;
-  }
-}
+
+    // If it has changed, save it and highlight it as the current step
+    if (currentStep !== newStep) {
+      setCurrentStep(newStep);
+      fire('zoom_step', steps[newStep]);
+    }
+  };
+
+  const toggleSize = () => {
+    setExpanded(!expanded);
+    fire('move_mobile_bottom_ui', stepsRef.current.offsetHeight);
+  };
+
+  // On load, zoom on step 0, unexpanded
+  useEffect(
+    () => {
+      fire('move_mobile_bottom_ui', 105);
+      fire('zoom_step', steps[0]);
+    },
+    []
+  );
+
+  // When expanded is modified, move the map's bottom UI
+  useEffect(
+    () => {
+      fire('move_mobile_bottom_ui', stepsRef.current.offsetHeight);
+    },
+    [expanded]
+  );
+
+  // When currentStep is modified, zoom on it
+  useEffect(
+    () => {
+      fire('zoom_step', steps[currentStep]);
+    },
+    [currentStep]
```

Comment:
Same for `steps` here. And this is normal, if for any reason the `steps` prop changes, you'll want to trigger the effect.